recipes/z3/all/conanfile.py

```diff
@@ -66,64 +73,59 @@ def requirements(self):
         elif self.options.multiprecision == "internal":
             pass
 
-    def source(self):
-        get(self, **self.conan_data["sources"][self.version],
-            destination=self._source_subfolder, strip_root=True)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["Z3_USE_LIB_GMP"] = self.options.multiprecision != "internal"
-        self._cmake.definitions["Z3_USE_LIB_MPIR"] = self.options.multiprecision == "mpir"
-        self._cmake.definitions["SINGLE_THREADED"] = not self.options.multithreaded
-        self._cmake.definitions["Z3_BUILD_LIBZ3_SHARED"] = self.options.shared
-        self._cmake.definitions["Z3_INCLUDE_GIT_HASH"] = False
-        self._cmake.definitions["Z3_INCLUDE_GIT_DESCRIBE"] = False
-        self._cmake.definitions["Z3_ENABLE_EXAMPLE_TARGETS"] = False
-        self._cmake.definitions["Z3_BUILD_DOCUMENTATION"] = False
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
-
-    @property
-    def _compilers_minimum_version(self):
-        return {
-            "gcc": "7",
-            "Visual Studio": "15.7",
-            "clang": "5",
-            "apple-clang": "10",
-        }
-
     def validate(self):
         if Version(self.version) >= "4.8.11":
-            if self.settings.compiler.get_safe("cppstd"):
-                tools.check_min_cppstd(self, "17")
-            compiler = self.settings.compiler
-            min_version = self._compilers_minimum_version\
-                .get(str(compiler), False)
-            if min_version:
-                if Version(compiler.version) < min_version:
-                    raise ConanInvalidConfiguration(
-                        f"{self.name} requires C++17, which {compiler} {compiler.version} does not support.")
+            if self.info.settings.compiler.cppstd:
+                check_min_cppstd(self, self._minimum_cpp_standard)
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version:
+                if Version(self.info.settings.compiler.version) < minimum_version:
+                    raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
             else:
                 self.output.info(
                     f"{self.name} requires C++17. Your compiler is unknown. Assuming it supports C++17.")
+        if is_msvc(self) and self.info.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
```

Comment:
Isn't `destination=self.source_folder` the default?

Reply:
The default is `"."` (https://docs.conan.io/en/latest/reference/conanfile/tools/files/downloads.html#conan-tools-files-get)

Reply:
Please, keep destination as an explicit parameter, it's easier for reviewers and for who reads the recipe for the first time.

Reply:
Can we resolve this ?

Reply:
Yes.